docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -128,3 +130,416 @@ class CustomerReader extends SprykerCustomerReader implements CustomerReaderInte
+
+```php
+...
+use Spryker\Zed\ProductPageSearch\ProductPageSearchDependencyProvider;
+...
+class CustomerAccessBusinessFactory implements SprykerCustomerAccessBusinessFactory
+{
+    /**
+    * @return \Pyz\Zed\CustomerAccess\Business\PropelFacadeInterface
+    * @throws \Spryker\Zed\Kernel\Exception\Container\ContainerKeyNotFoundException
+    */
+    public function getPropelFacade(): PropelFacadeInterface
+    {
+        return $this->getProvidedDependency(ProductPageSearchDependencyProvider::QUERY_CONTAINER_CATEGORY);
+    }
+}
+```
+
+### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+Please avoid usage of Spryker/Zed/ProductPageSearch/ProductPageSearchDependencyProvider/ProductPageSearchDependencyProvider::QUERY_CONTAINER_CATEGORY in Pyz/Zed/CustomerAccess/Business/CustomerAccessBusinessFactory
+------------------------------------------------------------------------------------------------------------------------
+```
+
+### Resolving the error
+
+The solution for the error is the same as in `Resolving the error`[PrivateApi:Facade](#privateapi-facade)
+
+
+## PrivateApi:DependencyInBusinessModel
+
+Business models on the project level should avoid the usage of private API from the Core level.
+
+### Example of code that causes an upgradability error
+
+`CustomerAccessFilter` uses as dependency `CustomerAccessReaderInterface` that was declared on the vendor level.
+
+```php
+...
+class CustomerAccessFilter implements CustomCustomerAccessFilter
+{
+    /**
+    * @param \Pyz\Zed\CustomerAccess\CustomerAccessConfig $customerAccessConfig
+    * @param \Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessReaderInterface $customerAccessReader
+    */
+    public function __construct(CustomerAccessConfig $customerAccessConfig, CustomerAccessReaderInterface $customerAccessReader)
+    {
+        $this->customerAccessReader = $customerAccessReader;
+        $this->customerAccessConfig = $customerAccessConfig;
+    }
+}
+```
+
+### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+PrivateApi:DependencyInBusinessModel "Please avoid usage of Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessReaderInterface in Pyz\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessFilter"
+------------------------------------------------------------------------------------------------------------------------
+```
+
+### Resolving the error
+
+The solution for the error is the same as in `Resolving the error`[PrivateApi:Facade](#privateapi-facade)
+
+
+## PrivateApi:ObjectInitialization
+
+Business models on the project level should avoid creating private API from the Core level.
```

Comment:
Is `creating` supposed to be `using`? If so, we need a better explanation, a bit confusing.

Reply:
in this case `creating` is close to `Initialization` `building` `new Object`. 

What about the version? 

Business models on the project level should avoid initializing private API from the Core level.